boot/boot_serial/src/boot_serial.c

```diff
@@ -313,6 +317,31 @@ static off_t erase_range(const struct flash_area *fap, off_t start, off_t end)
 }
 #endif
 
+static uint32_t next_offset(uint32_t curr_off, uint32_t img_size)
+{
+#ifndef MCUBOOT_BOOT_SERIAL_IMG_UPLOAD_STALLING
+    (void)img_size;
+#else
+    static uint32_t img_upload_stalling_timeout;
+
+    if (!img_upload_stalling) {
+        if (curr_off == img_size) {
+            img_upload_stalling = true;
+            img_upload_stalling_timeout =
+                k_uptime_get_32() + MCUBOOT_BOOT_SERIAL_IMG_UPLOAD_STALLING_TIMEOUT
```

Comment:
This is strange without the semicolon. Why not just have the number in the macro and use it like a regular expression?

Reply:
OK.... so this is programmers error and the only reason this even compiles is the other programmers error you have pointed out here: https://github.com/mcu-tools/mcuboot/pull/1387/files#r905364351.

Reply:
Fixed